### PR TITLE
Ensure that requires clauses are included in KCFG target states

### PR DESCRIPTION
**Pull request by ehildenb**
I was investigating a proof that is working with the legacy prover on KEVM but failing on the RPC prover, specifically [this one](https://github.com/runtimeverification/evm-semantics/blob/master/tests/specs/erc20/ds/transferFrom-failure-2-c-spec.k).

This proof was making it to the end of symbolic execution with the RPC prover very quickly (~ 3 minutes), and then spending half an hour on the implication check and eventually failing it. I minimized the example as much as possible, and included it as a test in this PR. It turns out, it cannot complete the implication check because the constraints from the requires clause are not also included in the target state here.

This PR adds this test, which also times out after a long time spent trying to prove the final implication (before these changes), then adds the fix (which is just to include the requires clause constraints in the target state of KCFGs too). Then the test passes in 19s (including kompilation time).

### CODE REVIEW

**Comment by ehildenb** [comment]
I will measure how many more KEVM proofs pass now with this change, and post results. These now passing testes will be included in https://github.com/runtimeverification/evm-semantics/pull/2067.

**Comment by ehildenb** [comment]
We can enable 35 more tests on KEVM with this change.

File: src/tests/integration/k-files/mini-kevm.k [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    syntax Int ::= keccak ( Bytes ) [function, total, no-evaluators, smtlib(keccak)]
+
+    syntax Int ::= #hashedLocation( String , Int , IntList ) [function, klabel(hashLoc), smtlib(hashLoc)]
+    rule #hashedLocation(_LANG, BASE, .IntList      ) => BASE
+    rule #hashedLocation( LANG, BASE, OFFSET OFFSETS) => #hashedLocation(LANG, #hashedLocation(LANG, BASE, OFFSET .IntList), OFFSETS) requires OFFSETS =/=K .IntList
+
+    rule #hashedLocation("Vyper",    BASE, OFFSET .IntList) => keccak(#bufStrict(32, BASE)   +Bytes #bufStrict(32, OFFSET)) requires #rangeUInt(256, BASE) andBool #rangeUInt(256, OFFSET)
+    rule #hashedLocation("Solidity", BASE, OFFSET .IntList) => keccak(#bufStrict(32, OFFSET) +Bytes #bufStrict(32, BASE))   requires #rangeUInt(256, BASE) andBool #rangeUInt(256, OFFSET) [simplification]
+    rule #hashedLocation("Array",    BASE, OFFSET .IntList) => keccak(#bufStrict(32, BASE))  +Word OFFSET                   requires #rangeUInt(256, BASE) andBool #rangeUInt(256, OFFSET)
```
Comment (tothtamas28) [inline src/tests/integration/k-files/mini-kevm.k:60]: Cases for `"Vyper"` and `"Array"` can probably be removed.

**Review by tothtamas28 — APPROVED** [review]

File: src/tests/integration/k-files/mini-kevm.k [hunk 2]
```diff
@@ -0,0 +1,62 @@
+requires "domains.md"
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply src/tests/integration/k-files/mini-kevm.k:1]: I guess it's probably not, it's just leftover as part of minimization process.

[the remark this answers is not in the document]
Comment (ehildenb) [reply src/tests/integration/k-files/mini-kevm.k]: Removed.

File: src/tests/integration/k-files/mini-kevm.k (hunk 1 above)
Comment (ehildenb) [reply src/tests/integration/k-files/mini-kevm.k]: Removed.

**Pull request merged**